### PR TITLE
Add live area game screen.

### PR DESCRIPTION
**Pull request by Zangetsu38**
- Add live area game screen.
- Add manual on live area game screen
- is also compatible on multi language if game have this information/file

# Live Area Result : 
- Persona 4 Golden: 
image

- A.O.T
image

- Danganronpa V3: Killing Harmony
image

- Adventures of Mana
image

- Persona 5: Dancing Star Night
image

# Manual Result: 

- Persona 4 Golden:
image
image

-A.O.T
image

- Danganronpa V3: Killing Harmony
image

### CODE REVIEW

**Comment by Zangetsu38** [comment]
Ready for review

File: vita3k/gui/src/gui.cpp [hunk 1]
```diff
@@ -329,6 +329,11 @@ void draw_ui(GuiState &gui, HostState &host) {
     if (gui.help_menu.about_dialog)
         draw_about_dialog(gui);
 
+    if (gui.live_area.live_area_dialog)
+        live_area_dialog(gui, host);
```
Comment (VelocityRa) [inline vita3k/gui/src/gui.cpp]:
```suggestion
        draw_live_area_dialog(gui, host);
```
for consistent naming

File: vita3k/gui/src/gui.cpp [hunk 2]
```diff
@@ -329,6 +329,11 @@ void draw_ui(GuiState &gui, HostState &host) {
...
+    if (gui.live_area.manual_dialog)
+        manual_dialog(gui, host);
```
Comment (VelocityRa) [inline vita3k/gui/src/gui.cpp]: ditto

File: vita3k/gui/src/live_area_dialog.cpp [hunk 3]
```diff
@@ -0,0 +1,715 @@
...
+
+    ImGui::SetCursorPos(pos_scal_search);
+    if (gui.live_area_contents[host.io.title_id].find(search) != gui.live_area_contents[host.io.title_id].end())
+        ImGui::Image(gui.live_area_contents[host.io.title_id][search], widget_scal_size);
+    else
+        ImGui::Button("Search", BUTTON_SIZE);
+    if (ImGui::IsItemClicked(0)) {
+        std::ostringstream link;
+        std::replace(host.game_title.begin(), host.game_title.end(), ' ', '+');
+        std::string search_url = "http://www.google.fr/search?q=" + host.game_title;
```
Comment (VelocityRa) [inline vita3k/gui/src/live_area_dialog.cpp]:
```suggestion
        std::string search_url = "http://www.google.com/search?q=" + host.game_title;
```

File: vita3k/gui/src/live_area_dialog.cpp [hunk 4]
```diff
@@ -0,0 +1,715 @@
...
+        ImGui::SetCursorPos(pos_scal_manual);
+        if (gui.live_area_contents[host.io.title_id].find(manual) != gui.live_area_contents[host.io.title_id].end())
+            ImGui::Image(gui.live_area_contents[host.io.title_id][manual], widget_scal_size);
+        else
+            ImGui::Button("Manual", BUTTON_SIZE);
+        if (ImGui::IsItemClicked(0)) {
+            if (init_manual(gui, host))
+                gui.live_area.manual_dialog = true;
+            else
+                LOG_ERROR("Error open Manual");
```
Comment (VelocityRa) [inline vita3k/gui/src/live_area_dialog.cpp]:
```suggestion
                LOG_ERROR("Error opening Manual");
```

File: vita3k/gui/src/manual_dialog.cpp [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+        ImGui::SetCursorPos(ImVec2(10.0f * scal.x, display_size.y - (40.0f * scal.y)));
+        if ((!hiden_button && ImGui::Button("<", BUTTON_SIZE)) || ImGui::IsKeyPressed(SDL_SCANCODE_LEFT) || (wheel_counter == 1))
+            --current_page;
+    }
+    if (!hiden_button) {
+        ImGui::SameLine();
+        ImGui::SetCursorPos(ImVec2(display_size.x / 2 - (40.0f * scal.x), display_size.y - (40.0f * scal.y)));
+        const std::string slider = fmt::format("{}/{}", current_page + 1, manual_page_count[host.io.title_id]);
+        if (ImGui::Button(slider.c_str(), BUTTON_SIZE))
+            ImGui::OpenPopup("Slider Manual");
```
Comment (VelocityRa) [inline vita3k/gui/src/manual_dialog.cpp]:
```suggestion
            ImGui::OpenPopup("Manual Slider");
```

File: vita3k/modules/SceAvPlayer/SceAvPlayer.cpp [hunk 6]
```diff
@@ -39,108 +39,109 @@ constexpr bool CATCHUP_VIDEO_PLAYBACK = true;
...
-    };
-
-    struct SceAvPlayerFrameInfo {
-        Ptr<uint8_t> data;
-        uint32_t unknown;
-        uint64_t timestamp;
-        emu::SceAvPlayerStreamDetails stream_details;
-    };
-}
+typedef Ptr<void> (*SceAvPlayerAllocator)(void *arguments, uint32_t alignment, uint32_t size);
```
Comment (VelocityRa) [inline vita3k/modules/SceAvPlayer/SceAvPlayer.cpp]: put unrelated format changes like this (all of them below here) to another "all: Format project" commit

File: vita3k/gui/src/live_area_dialog.cpp (hunk 3 above)
Comment (Zangetsu38) [reply vita3k/gui/src/live_area_dialog.cpp]: oups lol

**Comment by Zangetsu38** [comment]
All change is done normally

File: vita3k/config/include/config/config.h [hunk 7]
```diff
@@ -32,6 +32,7 @@
     code(bool, "pstv-mode", false, pstv_mode)                                                           \
     code(bool, "show-gui", true, show_gui)                                                              \
     code(bool, "show-game-background", false, show_game_background)                                     \
+    code(bool, "show-live-area-screen", true, show_live_area_screen)                                    \
```
Comment (VelocityRa) [inline vita3k/config/include/config/config.h]: wrong commit

**Comment by VelocityRa** [comment]
Can you post a screenshot of all the text? Like the Manual dialogs.

**Comment by Zangetsu38** [comment]
Done

**Comment by Zangetsu38** [comment]
sorry, pushed fix, forget clean manual page list before it re launch again ^^

File: vita3k/gui/src/game_selector.cpp [hunk 8]
```diff
@@ -251,7 +262,30 @@ void draw_game_selector(GuiState &gui, HostState &host) {
...
+                    live_area_game(gui, host);
+                } else
+                    gui.game_selector.selected_title_id = game.title_id;
+            }
+            if (host.io.title_id == game.title_id) {
+                static const auto BUTTON_SIZE = ImVec2(60.f, 0.f);
+                ImGui::PushStyleColor(ImGuiCol_Text, GUI_COLOR_TEXT_MENUBAR_OPTIONS);
+                if (ImGui::BeginPopupModal("Live Area Error", nullptr, ImGuiWindowFlags_AlwaysAutoResize)) {
+                    ImGui::PopStyleColor();
+                    ImGui::TextColored(GUI_COLOR_TEXT, "Failed open Livea Area for:\n%s '%s'.\nDo you want launch it?", game.title_id.c_str(), game.title.c_str());
```
Comment (1whatleytay) [inline vita3k/gui/src/game_selector.cpp]: If you can't open live area, just launch it. No need for a popup I think.

File: vita3k/gui/src/gui.cpp [hunk 9]
```diff
@@ -153,12 +153,17 @@ void init_icons(GuiState &gui, HostState &host) {
...
+
+        const auto default_fw_icon{ fs::path(host.pref_path) / "vs0/data/internal/livearea/default/sce_sys/icon0.png" };
+        const auto default_icon{ fs::path(host.base_path) / "data/image/icon.png" };
+
+        if ((buffer.empty() && (fs::exists(default_fw_icon) || fs::exists(default_icon)))) {
             LOG_INFO("Default icon found for title {}, {}.", game.title_id, game.title);
-            std::ifstream image_stream(default_icon, std::ios::binary | std::ios::ate);
+            std::ifstream image_stream(default_fw_icon.string(), std::ios::binary | std::ios::ate);
+            if (!fs::exists(default_fw_icon))
+                std::ifstream image_stream(default_icon.string(), std::ios::binary | std::ios::ate);
```
Comment (1whatleytay) [inline vita3k/gui/src/gui.cpp]:
```suggestion
                image_stream = std::ifstream(default_icon.string(), std::ios::binary | std::ios::ate);
```

File: vita3k/gui/src/gui.cpp [hunk 10]
```diff
@@ -177,31 +182,26 @@ void init_icons(GuiState &gui, HostState &host) {
...
-            LOG_INFO("Game background found in template for title {}.", title_id);
-            vfs::read_app_file(buffer, host.pref_path, title_id, "sce_sys/livearea/contents/bg.png");
-            vfs::read_app_file(buffer, host.pref_path, title_id, "sce_sys/livearea/contents/bg0.png");
-        } else {
-            if (gui.current_background != gui.user_backgrounds[host.cfg.background_image])
-                gui.current_background = gui.user_backgrounds[host.cfg.background_image];
-            LOG_WARN("Game background not found for title {}.", title_id);
-            return;
-        }
+        if (gui.current_background != gui.user_backgrounds[host.cfg.background_image])
```
Comment (1whatleytay) [inline vita3k/gui/src/gui.cpp]:
```suggestion
```

File: vita3k/gui/src/imgui_impl_sdl_vulkan.cpp [hunk 11]
```diff
@@ -416,10 +416,22 @@ IMGUI_API void ImGui_ImplSdlVulkan_RenderDrawData(ImGui_VulkanState &state) {
     state.command_buffer.begin(begin_info);
```
Comment (1whatleytay) [inline vita3k/gui/src/imgui_impl_sdl_vulkan.cpp]:
```suggestion
 
    // clang-format off
```

File: vita3k/gui/src/imgui_impl_sdl_vulkan.cpp [hunk 12]
```diff
@@ -416,10 +416,22 @@ IMGUI_API void ImGui_ImplSdlVulkan_RenderDrawData(ImGui_VulkanState &state) {
...
+        0,
+        0,
+        0,
+        1,
+        0,
+        -1,
+        -1,
+        0,
+        1,
     };
```
Comment (1whatleytay) [inline vita3k/gui/src/imgui_impl_sdl_vulkan.cpp]:
```suggestion
    // clang-format on

```

File: vita3k/gui/src/live_area_dialog.cpp [hunk 13]
```diff
@@ -0,0 +1,694 @@
...
+    frame6,
+    frame7,
+    bg,
+    img,
+    gate,
+    search,
+    manual /*,
+    button,
+    refresh*/
+};
```
Comment (1whatleytay) [inline vita3k/gui/src/live_area_dialog.cpp]: Our enum style I think is:
```
enum LiveAreaContentsZone {
    BACKGROUND,
    FRAME1,
    FRAME2,
    FRAME3,
    ...
```

SoLikeThisForName and LIKE_THIS_FOR_ELEMENTS

File: vita3k/gui/src/live_area_dialog.cpp [hunk 14]
```diff
@@ -0,0 +1,694 @@
...
+            if (last_time[host.io.title_id][flip.first] == 0)
+                last_time[host.io.title_id][flip.first] = current_time();
+
+            while (last_time[host.io.title_id][flip.first] + flip.second < current_time()) {
+                last_time[host.io.title_id][flip.first] += flip.second;
+
+                if (gui.live_items[host.io.title_id][flip.first].find(bg) != gui.live_items[host.io.title_id][flip.first].end()) {
+                    if (current_item[host.io.title_id][flip.first] != int(gui.live_items[host.io.title_id][flip.first][bg].size()) - 1)
+                        ++current_item[host.io.title_id][flip.first];
+                    else if (current_item[host.io.title_id][flip.first] == int(gui.live_items[host.io.title_id][flip.first][bg].size() - 1))
```
Comment (1whatleytay) [inline vita3k/gui/src/live_area_dialog.cpp]:
```suggestion
                    else
```

The nature of else means you can drop this `if`.

File: vita3k/gui/src/live_area_dialog.cpp [hunk 15]
```diff
@@ -0,0 +1,694 @@
...
+                        current_item[host.io.title_id][flip.first] = 0;
+                } else if (gui.live_items[host.io.title_id][flip.first].find(img) != gui.live_items[host.io.title_id][flip.first].end()) {
+                    if (current_item[host.io.title_id][flip.first] != int(gui.live_items[host.io.title_id][flip.first][img].size()) - 1)
+                        ++current_item[host.io.title_id][flip.first];
+                    else if (current_item[host.io.title_id][flip.first] == int(gui.live_items[host.io.title_id][flip.first][img].size() - 1))
```
Comment (1whatleytay) [inline vita3k/gui/src/live_area_dialog.cpp]:
```suggestion
                    else
```

same here, i'm pretty sure both statements are exact opposites

File: vita3k/gxm/src/textures.cpp [hunk 16]
```diff
@@ -33,8 +33,6 @@ bool is_paletted_format(SceGxmTextureFormat src) {
 bool is_yuv_format(SceGxmTextureFormat src) {
     const auto base_format = get_base_format(src);
 
-    return base_format == SCE_GXM_TEXTURE_BASE_FORMAT_YUV420P2 ||
-        base_format == SCE_GXM_TEXTURE_BASE_FORMAT_YUV420P3 ||
-        base_format == SCE_GXM_TEXTURE_BASE_FORMAT_YUV422;
+    return base_format == SCE_GXM_TEXTURE_BASE_FORMAT_YUV420P2 || base_format == SCE_GXM_TEXTURE_BASE_FORMAT_YUV420P3 || base_format == SCE_GXM_TEXTURE_BASE_FORMAT_YUV422;
```
Comment (1whatleytay) [inline vita3k/gxm/src/textures.cpp]: this is a long line

why did clang format do this

File: vita3k/shader/src/translator/alu.cpp [hunk 17]
```diff
@@ -1416,7 +1412,7 @@ bool USSETranslatorVisitor::vdual(
     std::string disasm_str = fmt::format("{:016x}: ", m_instr);
 
     auto do_dual_op = [&](Opcode code, std::vector<Operand> &ops, Operand &dest,
-        Imm4 write_mask_dest, const DualOpInfo &code_info) {
+                          Imm4 write_mask_dest, const DualOpInfo &code_info) {
```
Comment (1whatleytay) [inline vita3k/shader/src/translator/alu.cpp]: all the other format stuff could probably do in its own commit if you can

thank you for cleaning up my mess 🙂

File: vita3k/gui/include/gui/functions.h [hunk 18]
```diff
@@ -56,7 +59,7 @@ void draw_reinstall_dialog(GenericDialogState *status);
 void draw_trophies_unlocked(GuiState &gui, HostState &host);
 void draw_perf_overlay(GuiState &gui, HostState &host);
 
-ImTextureID load_image(GuiState &gui, const char *data, const std::size_t size);
+ImTextureID load_image(GuiState &gui, const char *data, const int size);
```
[the remark this answers is not in the document]
Comment (Zangetsu38) [reply vita3k/gui/include/gui/functions.h]: For info i est it because i fix one warning, it convert size_t on int

File: vita3k/gui/src/game_selector.cpp (hunk 8 above)
Comment (Zangetsu38) [reply vita3k/gui/src/game_selector.cpp]: Hmm yes right

File: vita3k/gui/src/gui.cpp (hunk 10 above)
Comment (Zangetsu38) [reply vita3k/gui/src/gui.cpp]: No that is request

File: vita3k/gui/src/live_area_dialog.cpp (hunk 14 above)
Comment (Zangetsu38) [reply vita3k/gui/src/live_area_dialog.cpp]: No is needed

File: vita3k/gxm/src/textures.cpp (hunk 16 above)
Comment (Zangetsu38) [reply vita3k/gxm/src/textures.cpp]: No idea

File: vita3k/gui/include/gui/functions.h [hunk 19]
```diff
@@ -36,14 +36,17 @@ enum GenericDialogState {
...
-void delete_game(GuiState &gui, HostState &host, const std::string &title_id);
-void game_context_menu(GuiState &gui, HostState &host, bool *selected, const std::string &title_id);
+void delete_game(GuiState &gui, HostState &host);
+void game_context_menu(GuiState &gui, HostState &host);
 void get_game_titles(GuiState &gui, HostState &host);
 void get_modules_list(GuiState &gui, HostState &host);
 void init(GuiState &gui, HostState &host);
 void init_background(GuiState &gui, const std::string &image_path);
 void init_icons(GuiState &gui, HostState &host);
-void load_game_background(GuiState &gui, HostState &host, const std::string &title_id);
```
[the remark this answers is not in the document]
Comment (Zangetsu38) [reply vita3k/gui/include/gui/functions.h]: Drop it is just not needed now

File: vita3k/gui/src/gui.cpp (hunk 10 above)
Comment (1whatleytay) [reply vita3k/gui/src/gui.cpp]: LOL ok

Comment (Zangetsu38) [reply vita3k/gui/src/gui.cpp]: i same befàore, i want change only if is needed, master is already like that

File: vita3k/gui/src/game_selector.cpp (hunk 8 above)
Comment (Zangetsu38) [reply vita3k/gui/src/game_selector.cpp]: after thinks, is needed, no forget i using scroll and arrow for switch in game list, so if i using like you said, it launch game also during this switch by accident

File: vita3k/gui/src/gui.cpp (hunk 9 above)
Comment (Zangetsu38) [reply vita3k/gui/src/gui.cpp]: ah thx

File: vita3k/gui/src/game_selector.cpp (hunk 8 above)
Comment (1whatleytay) [reply vita3k/gui/src/game_selector.cpp]: Try to make it work I think.

Comment (Zangetsu38) [reply vita3k/gui/src/game_selector.cpp]: no, i said i can't, nbot works, this in used for everyy switch game, not only for click on it, so no want launch game by accident ^^

File: vita3k/config/include/config/config.h [hunk 20]
```diff
@@ -1,5 +1,5 @@
 // Vita3K emulator project
-// Copyright (C) 2018 Vita3K team
+// Copyright (C) 2020 Vita3K team
```
Comment (1whatleytay) [inline vita3k/config/include/config/config.h]: happy new year

if we aren't going to change the date for every file in the project its probably okay to leave it

File: vita3k/shader/src/usse_translator_entry.cpp [hunk 21]
```diff
@@ -37,11 +37,11 @@ using USSEMatcher = shader::decoder::Matcher<Visitor, uint64_t>;
 template <typename V>
 boost::optional<const USSEMatcher<V> &> DecodeUSSE(uint64_t instruction) {
     static const std::vector<USSEMatcher<V>> table = {
-    // clang-format off
-    #define INST(fn, name, bitstring) shader::decoder::detail::detail<USSEMatcher<V>>::GetMatcher(fn, name, bitstring)
+#define INST(fn, name, bitstring) shader::decoder::detail::detail<USSEMatcher<V>>::GetMatcher(fn, name, bitstring)
+        // clang-format off
```
Comment (1whatleytay) [inline vita3k/shader/src/usse_translator_entry.cpp]: What in the world happened here?

There's a comment disabling the formatter, and it had the guts to actually move it! Probably best if we unstage it.

Comment (Zangetsu38) [reply vita3k/shader/src/usse_translator_entry.cpp]: Set this before get problèm, it get all time change with run clang format

File: vita3k/config/include/config/config.h (hunk 20 above)
Comment (Zangetsu38) [reply vita3k/config/include/config/config.h]: I have change only on file changed ^^
